trick_sims/SIM_singlerigidbody/models/src/body.cpp

```diff
@@ -0,0 +1,353 @@
+*****************************************************************************/
+void BODY::calctorque() {
+
+	double torque_force[3];
+
+	V_CROSS(torque_force, POS_FORCE, FORCE_INIT);
+	MxV(torque, rotate, torque_force);
+
+};
+
+/****************************************************************************
+FUNCTION: BODY::calcforce_remain()
+PURPOSE: (Calculate remaining force wrt inertial frame)
+****************************************************************************/
+void BODY::calcforce_remain() {
+	
+	double cross1[3];
+	double cross2[3];
+
+	V_CROSS(cross1, omega, CM);
+	V_CROSS(cross2, omega, cross1);
+	V_SCALE(force_remain, cross2, mass);
+
+};
+
+
+/***************************************************************************
+FUNCTION: BODY::calctorque_remain()
+PURPOSE: (Calculate remaining torque wrt inertial frame)
+***************************************************************************/
+void BODY::calctorque_remain() {
+	
+	double I_w[3]; 
+
+	MxV(I_w, inertia_matrix, omega);
+	V_CROSS(torque_remain, omega, I_w);
+	
+
+};
+
+/***************************************************************************
+FUNCTION: BODY::eq_setup()
+PURPOSE: (Establishes b column vector for final eq M*a=b)
+***************************************************************************/
+void BODY::eq_setup() {
+	
+	double remainder_force[3];
+ 	double remainder_torque[3];	
+
+	// Subtracts remaining from force & torque to get vec_b. 
+	V_SUB(remainder_force, force, force_remain);
+	V_SUB(remainder_torque, torque, torque_remain);
+		
+	for(int i = 0; i<3; i++){
+		vec_b[i] = remainder_force[i];
+	}
+	
+	for(int i = 3; i<6; i++){
+		vec_b[i] = remainder_torque[i-3];
+	}
+
+};
+
+/**************************************************************************
+FUNCTION: BODY::eq_solver()
+PURPOSE: (Performs choleski decomposition and computes linear and angular
+          accelerations)
+**************************************************************************/
+
+void BODY::eq_solver() {
+	
+	// Solving a = b * M^-1
+	
+	// Temporary vector for Choleski 
+	double tmp_vec[6] = {0.0, 0.0, 0.0, 0.0, 0.0, 0.0};
+
+	// Dynamic memory set up
+	double **mat_mass_dyn;
+	double **mat_L;
+	mat_L = new double*[6];
+	mat_mass_dyn = new double*[6];
+	
+	for(int i = 0; i<6; i++)
+	{
+		mat_mass_dyn[i] = mat_mass[i];  
+		mat_L[i] = new double[6];
+	}
+	
+		
+	dLU_Choleski(mat_mass_dyn, mat_L, tmp_vec, 6, vec_b, vec_a, 0);
+	
+	// Acceleration output
+	for(int i = 0; i<3; i++)
+		acc[i] = vec_a[i];
+
+	// Angular accleration output	
+	for(int i = 0; i<3; i++)
+		omegaDot[i] = vec_a[i + 3];
+
+	// Delete dynamic memory
+	delete[] mat_mass_dyn;
+	delete[] mat_L;
+
+};
+
+/***************************************************************************
+FUNCTION: BODY::derivative()
+PURPOSE: (Calls all functions in desired order for calculations)
+***************************************************************************/
+void BODY::derivative() {
+
+	mass_matrix();
```

Comment:
see comment above about moving mass matrix calculation out of derivative job call